vizro-core/src/vizro/_vizro.py

```diff
@@ -46,41 +58,18 @@ def __init__(self, **kwargs):
         )
 
-        # Include Vizro assets (in the static folder) as external scripts and stylesheets. We extend self.dash.config
-        # objects so the user can specify additional external_scripts and external_stylesheets via kwargs.
-        vizro_assets_folder = Path(__file__).with_name("static")
-        requests_pathname_prefix = self.dash.config.requests_pathname_prefix
-        # Exclude vizro/css/figures.css since these are distributed through the vizro._css_dist mechanism.
-        # In future we will probably handle all assets this way and none of this code will be required.
-        vizro_css = [
-            requests_pathname_prefix + path
-            for path in self._get_external_assets(vizro_assets_folder, "css")
-            if path != "vizro/css/figures.css"
-        ]
-
-        # Ensure vizro-bootstrap.min.css is loaded in first to allow overwrites
-        vizro_css.sort(key=lambda x: not x.endswith("vizro-bootstrap.min.css"))
-
-        vizro_js = [
-            {"src": requests_pathname_prefix + path, "type": "module"}
-            for path in self._get_external_assets(vizro_assets_folder, "js")
-        ]
-        self.dash.config.external_stylesheets.extend(vizro_css)
-        self.dash.config.external_scripts.extend(vizro_js)
-
-        # Serve all assets (including files other than css and js) that live in vizro_assets_folder at the
-        # route /vizro. Based on code in Dash.init_app that serves assets_folder. This respects the case that the
-        # dashboard is not hosted at the root of the server, e.g. http://www.example.com/dashboard/vizro.
-        routes_pathname_prefix = self.dash.config.routes_pathname_prefix
-        blueprint_prefix = routes_pathname_prefix.replace("/", "_").replace(".", "_")
-        self.dash.server.register_blueprint(
-            flask.Blueprint(
-                f"{blueprint_prefix}vizro_assets",
-                self.dash.config.name,
-                static_folder=vizro_assets_folder,
-                static_url_path=routes_pathname_prefix + STATIC_URL_PREFIX,
-            )
-        )
+        # Add static assets that were not already included in the library. These are registered only when Vizro() is
+        # called, i.e. when Vizro is used as a framework.
+        for path in set(VIZRO_ASSETS_PATH.rglob("*")) - _library_css_files - _library_js_files:
+            if path.suffix == ".css":
+                self.dash.css.append_css(_make_resource_spec(path))
+            elif path.suffix == ".js":
+                self.dash.scripts.append_script(_make_resource_spec(path))
+            else:
+                # map files and fonts and images. These are treated like scripts since this is how Dash handles them.
+                # This adds paths to self.dash.registered_paths so that they can be accessed without throwing an
+                # error in dash._validate.validate_js_path.
+                self.dash.scripts.append_script(_make_resource_spec(path))
```

Comment:
I 👍  keeping this separate although the code block under the `elif` and `else` is the same.

Reply:
Haha yes, I spent a while thinking about which I preferred and decided this would be clearest so glad you agree!